sites-available/matomo.conf

```diff
@@ -2,11 +2,10 @@ server {
     listen [::]:80; # remove this if you don't want Matomo to be reachable from IPv6
     listen 80;
     server_name matomo.example.com;
-    location / {
-        # Redirect all HTTP requests to HTTPS with a 301 Moved Permanently response.
-        return 301 https://$host$request_uri;
-    }
+    # Redirect all HTTP requests to HTTPS with a 301 Moved Permanently response.
+    return 301 https://$server_name$request_uri;
```

Comment:
:+1:

Reply:
Just for your information, I applied this change in https://github.com/matomo-org/matomo-nginx/commit/b74e721746f3f70a8cfffe911996c9e81d5ae8dd, but reverted it in https://github.com/matomo-org/matomo-nginx/commit/7035d4ff1076e34ef34a384e0963b519713698d2 as it causes troubles when one adds a second location block there (e.g. for let's encrypt) and doesn't notice that this is outside a block.

Reply:
Thanks for the hint, that makes sense. Maybe a comment on this would be helpful to other users, that $host have security issues and $server_name should be used when only one URL is in use ?

Reply:
Ah, thanks. I didn't notice that this was also changed. 
Can you quickly describe me a use case where this would make it more secure?
If I understand it correctly $host takes the Host: header and $server_name the first domain specified in the nginx block.

So as long people don't use Matomo as the default_server it shouldn't matter as the Host as to match anyway. But if they do (which they shouldn't), an attacker could fake the $host ,right?